logic/resolvable/Concludable.java

```diff
@@ -433,30 +509,61 @@ public Has asHas() {
 
         @Override
         AlphaEquivalence alphaEquals(Has that) {
-            return constraint().alphaEquals(that.constraint());
+            return has().alphaEquals(that.has()).validIfAlphaEqual(isa, that.isa).validIfAlphaEqual(values, that.values);
         }
 
-        private static Set<Constraint> core(@Nullable IsaConstraint isaConstraint, Set<ValueConstraint<?>> valueConstraints) {
-            Set<Constraint> c = new HashSet<>(valueConstraints);
-            if (isaConstraint != null) c.add(isaConstraint);
-            return c;
-        }
     }
 
-    public static class Isa extends Concludable<IsaConstraint> {
+    /**
+     * Isa handles concludable these concludable patterns, where the owner of the IsaConstraint is not already the
+     * subject of a Has or Relation Concludable:
+     * `{ $x isa person; }`
+     * `{ $a isa age; $a = 30; }`
+     * `{ $a isa age; $a > 5; $a < 30; }`
+     */
+    public static class Isa extends Concludable {
+
+        private final IsaConstraint isa;
+        private final Set<ValueConstraint<?>> values;
 
-        public Isa(final IsaConstraint constraint, Set<ValueConstraint<?>> value) {
-            super(ConstraintCopier.copyConstraint(constraint), new HashSet<>(value));
+        private Isa(Conjunction conjunction, IsaConstraint isa, Set<ValueConstraint<?>> values) {
+            super(conjunction);
+            this.isa = isa;
+            this.values = values;
+        }
+
+        public static Isa of(IsaConstraint isa, Set<ValueConstraint<?>> values, Set<LabelConstraint> labelConstraints) {
+            Conjunction.Cloner cloner = Conjunction.Cloner.cloneExactly(labelConstraints, values, isa);
+            return new Isa(cloner.conjunction(), cloner.getClone(isa).asThing().asIsa(), new HashSet<>(
+                    Iterators.iterate(values).map(cloner::getClone).map(c -> c.asThing().asValue()).toSet()));
+        }
+
+        public IsaConstraint isa() {
+            return isa;
         }
 
         @Override
+        public Set<Constraint> concludableConstraints() {
+            Set<Constraint> constraints = new HashSet<>();
+            constraints.add(isa);
+            constraints.addAll(equalsConstraints(values));
+            return constraints;
+        }
+
+        @Override
+        ResourceIterator<Unifier> unify(Rule.Conclusion conclusion, ConceptManager conceptMgr) {
+            if (conclusion.isIsa()) return unify(conclusion.asIsa(), conceptMgr);
+            return Iterators.empty();
```

Comment:
Any rule conclusion that can infer an `isa` implements the `isa` interface: `Relation`, `ExplicitHas`, `Isa`, so here it's this simple.